console/console-init/ui/src/modules/address-space/utils/utils.ts

```diff
@@ -21,4 +24,72 @@ const getHeaderForDeleteDialog = (selectedItems: any[]) => {
   return header;
 };
 
-export { getDetailForDeleteDialog, getHeaderForDeleteDialog };
+const isMessgaingProjectConfigurationValid = (
+  messagingProject: IMessagingProject
+) => {
+  if (messagingProject) {
+    const { name, namespace, type, plan, authService } = messagingProject;
+    if (
+      name &&
+      name.trim() !== "" &&
+      namespace &&
+      namespace.trim() !== "" &&
+      type &&
+      type.trim() !== "" &&
+      plan &&
+      plan.trim() !== "" &&
+      authService &&
+      authService.trim() !== ""
+    ) {
+      return true;
+    }
+    return false;
+  }
+};
+
+const isEnabledCertificateStep = (messagingProject: IMessagingProject) => {
+  if (messagingProject.customizeEndpoint === true) {
+    if (messagingProject.tlsCertificate !== TlsCertificateType.UPLOAD_CERT) {
+      return true;
+    } else {
+      if (
+        messagingProject.tlsCertificate === TlsCertificateType.UPLOAD_CERT &&
+        messagingProject.privateKey &&
+        messagingProject.privateKey.trim() !== "" &&
+        messagingProject.certValue &&
+        messagingProject.certValue?.trim() !== ""
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  } else {
+    return false;
+  }
+};
+
+const isMessgaingProjectValid = (messagingProject: IMessagingProject) => {
```

Comment:
Simple version of this function. can you try it.

const isMessgaingProjectValid = (messagingProject: IMessagingProject) => {
  if (isMessgaingProjectConfigurationValid(messagingProject) &&
     (messagingProject.customizeEndpoint === true ||
        messagingProject.protocols &&
        messagingProject.protocols.length > 0 &&
        messagingProject.tlsCertificate &&
        messagingProject.tlsCertificate.trim() !== "" 
     )
  ) {
     return true;
  }
  return false;
};